### PR TITLE
refactor: Make mock fetcher generally available

### PR DESCRIPTION
**Pull request by TerryHowe**
**What this PR does / why we need it**:

I wanted to get a feel for how people felt about making a mock fetcher as a test utility, so I created this small PR.

### CODE REVIEW

File: internal/testutils/fetcher.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+func (mf *MockFetcher) pushImage(subject *ocispec.Descriptor, mediaType string, config ocispec.Descriptor, layers ...ocispec.Descriptor) ocispec.Descriptor {
+	manifest := ocispec.Manifest{
+		MediaType: mediaType,
+		Subject:   subject,
+		Config:    config,
+		Layers:    layers,
+	}
+	manifestJSON, err := json.Marshal(manifest)
+	if err != nil {
+		mf.t.Fatal(err)
```
Comment (qweeah) [inline internal/testutils/fetcher.go:74]: Anyway we can exclude this file or this package? Test code should not be calculated into coverage counting.

**Review by qweeah — CHANGES_REQUESTED** [review]
Well, I got to confess mocking a fetcher is toll and the test code is that bad, but you haven't met the mocked registry service yet :)

File: internal/testutils/fetcher.go
Comment (qweeah) [inline internal/testutils/fetcher.go:1]: Need comment doc for all public functions.

Comment (TerryHowe) [reply internal/testutils/fetcher.go:1]: Done.

File: internal/testutils/fetcher.go (hunk 1 above)
Comment (TerryHowe) [reply internal/testutils/fetcher.go:76]: I'm not sure how exclude this from coverage. All the ways I'm seeing to do that are not great.

Comment (TerryHowe) [reply internal/testutils/fetcher.go:76]: Okay, that seemed to work. No coverage on this.

**Review by qweeah — APPROVED** [review]
LGTM

**Pull request merged**